tests/5.1/allocate/test_target_calloc.c

```diff
@@ -0,0 +1,80 @@
+//===------------------------ test_calloc_host.c --------------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// Tests the omp_calloc routine to make sure memory is properly allocated
+// on the host and is zero initialized. Allocates an array and makes sure
+// each value is set to 0 to start. Then adds i to each index and validates the
+// changes persisted and were written correctly.
+//
+//===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int test_omp_target_calloc() {
+
+  int errors = 0;
+
+  #pragma omp target map(tofrom: errors) uses_allocators(omp_default_mem_alloc)
+  {
+
+    int *x;
+
+    x = (int *)omp_calloc(64, N*sizeof(int), omp_default_mem_alloc);
+
+    if (x == NULL) { 
+      OMPVV_ERROR("omp_calloc returned null"); 
+      return (1); 
```

Comment:
GCC complains here:
```
test_target_calloc.c:33:14: error: invalid branch to/from OpenMP structured block
```
(However, GCC does accept `exit(1);` or `abort();`.)

Likewise with Clang:
```
test_target_calloc.c:35:7: error: cannot return from OpenMP region
```

From the OpenMP spec:
* _structured block_ — For C/C++, an executable statement, possibly compound, with a single entry at the9
top and a single exit at the bottom, or an OpenMP construct. [...]